ARCs/arc-0020.md

```diff
@@ -0,0 +1,870 @@
+---
+arc: 0020
+title: Smart ASA
+description: An ARC for an ASA controlled by an Algorand Smart Contract
+author:
+  - Cosimo Bassi (@cusma)
+  - Adriano Di Luzio (@aldur)
+  - John Jannotti (@jannotti)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): Interface
+created: 2022-04-27
+requires: ARC-4
+---
+
+## Abstract
+
+A "Smart ASA" is an Algorand Standard Asset (ASA) controlled by a Smart Contract
+that exposes methods to create, configure, transfer, freeze, and destroy the
+asset.
+
+This ARC defines the ABI interface of such Smart Contract, the required
+metadata, and suggests a reference implementation.
+
+## Motivation
+
+The Algorand Standard Asset (ASA) is an excellent building block for on-chain
+applications. It is battle-tested and widely supported by SDKs, wallets, and
+dApps.
+
+However, the ASA lacks in flexibility and configurability. For instance, once
+issued it can't be re-configured (its unit name, decimals, maximum supply).
+Also, it is freely transferable (unless frozen). This prevents developers from
+specifying additional business logic to be checked while transferring it (think
+of royalties or [vesting](https://en.wikipedia.org/wiki/Vesting)).
+
+Enforcing transfer conditions require freezing the asset and transferring it
```

Comment:
... and it's also not supported by any marketplaces and also requires a great deal of trust to be placed in the creator of the asset who could easily rug pull the assets.